### PR TITLE
Scanbuild warning and error fixes

### PR DESCRIPTION
**Pull request by akallabeth**

### CODE REVIEW

File: libfreerdp/primitives/prim_colors_opt.c [hunk 1]
```diff
@@ -1169,8 +1169,8 @@ static pstatus_t sse2_RGBToRGB_16s8u_P3AC4R(
     UINT32 DstFormat,
     const prim_size_t* roi)
 {
-	if (((UINT64)pSrc[0] & 0x0f) || ((UINT64)pSrc[0] & 0x0f) || ((UINT64)pSrc[0] & 0x0f) ||
-	    (srcStep & 0x0f) || ((UINT64)pDst & 0x0f) || (dstStep & 0x0f))
+	if (((ULONG_PTR)pSrc[0] & 0x0f) || ((ULONG_PTR)pSrc[0] & 0x0f) || ((ULONG_PTR)pSrc[0] & 0x0f) ||
```
Comment (hendwolt) [inline libfreerdp/primitives/prim_colors_opt.c]: Still 3 times ((ULONG_PTR)pSrc[0] & 0x0f) ?

Comment (akallabeth) [inline libfreerdp/primitives/prim_colors_opt.c]: @hendwolt This is fixing the warning on my ubuntu 16.04 32bit VM (as opposed to the old cast to ```UINT64```)

File: libfreerdp/codec/color.c [hunk 2]
```diff
@@ -84,17 +84,12 @@ BOOL freerdp_image_copy_from_monochrome(BYTE* pDstData, UINT32 DstFormat,
 {
 	UINT32 x, y;
 	BOOL vFlip;
-	UINT32 nDstPad;
 	UINT32 monoStep;
-	UINT32 dstBitsPerPixel;
-	UINT32 dstBytesPerPixel;
-	dstBitsPerPixel = GetBitsPerPixel(DstFormat);
-	dstBytesPerPixel = GetBytesPerPixel(DstFormat);
+	const UINT32 dstBytesPerPixel = GetBytesPerPixel(DstFormat);
```
Comment (hardening) [inline libfreerdp/codec/color.c]: there's no point to have a const here

File: libfreerdp/core/gateway/tsg.c [hunk 3]
```diff
@@ -210,25 +209,24 @@ BOOL TsProxyCreateTunnelWriteRequest(rdpTsg* tsg, PTSG_PACKET tsgPacket)
 		*((UINT32*) &buffer[20]) = packetVersionCaps->numCapabilities; /* NumCapabilities (4 bytes) */
 		*((UINT16*) &buffer[24]) = packetVersionCaps->majorVersion; /* MajorVersion (2 bytes) */
 		*((UINT16*) &buffer[26]) = packetVersionCaps->minorVersion; /* MinorVersion (2 bytes) */
-		*((UINT16*) &buffer[28]) = packetVersionCaps->quarantineCapabilities; /* QuarantineCapabilities (2 bytes) */
+		*((UINT16*) &buffer[28]) =
+		    packetVersionCaps->quarantineCapabilities; /* QuarantineCapabilities (2 bytes) */
 		/* 4-byte alignment (30 + 2) */
```
Comment (hardening) [inline libfreerdp/core/gateway/tsg.c]: sorry to bother but these reformatting are clearly against readability

File: libfreerdp/codec/color.c (hunk 2 above)
Comment (akallabeth) [inline libfreerdp/codec/color.c]: @hardening Just a precaution against accidental assignment

File: libfreerdp/primitives/prim_colors_opt.c (hunk 1 above)
Comment (hendwolt) [inline libfreerdp/primitives/prim_colors_opt.c]: It's not about the cast warning, but checking the same value (pSrc[0]) 3 times.
Shouldn't it be:
((ULONG_PTR)pSrc[0] & 0x0f) || ((ULONG_PTR)pSrc[1] & 0x0f) || ((ULONG_PTR)pSrc[2] & 0x0f)

**Pull request merged**